src/wallet/wallet.cpp

```diff
@@ -1532,7 +1531,16 @@ bool CWallet::SelectCoinsMinConf(int64_t nTargetValue, unsigned int nSpendTime,
     vector<pair<int64_t, pair<const CWalletTx*,unsigned int> > > vValue;
     int64_t nTotalLower = 0;
 
-    random_shuffle(vCoins.begin(), vCoins.end(), GetRandInt);
+    // A wrapper for GetRandInt to satisfy std::shuffle's UniformRandomBitGenerator requirements.
+    // TODO: Port FastRandomContext and replace this.
+    class {
+    public:
+        typedef int result_type;
+        static constexpr int min() { return 0; }
+        static constexpr uint64_t max() { return 0x7FFFFFFF; }
+        inline uint64_t operator()() noexcept { return GetRandInt(0x7FFFFFFF); }
+    } urbg;
+    std::shuffle(vCoins.begin(), vCoins.end(), urbg);
```

Comment:
This is a really ugly workaround, will address this in another PR.

Reply:
Ugh. Yep.

Reply:
@div72 The macOS tests are stalling on `std::shuffle()` here when `wallet_tests::coin_selection_tests()` calls `CWallet::SelectCoinsMinConf()`. Not sure why yet, but gdb seems to indicate that something in the linked Homebrew OpenSSL's libcrypto 1.1.0 is hitting an infinite loop when initializing some state for the `GetRandInt()` call. 

I'll try to come back to this tonight when I'll have more time. Really hope it's not an OpenSSL issue...

Reply:
Can't unresolve. Don't think I have any repo permissions.

Reply:
@cyrossignol Of course this had to be faulty, isn't it?

Are you sure the problem is OpenSSL's? Could it be related to https://stackoverflow.com/questions/22915325/avoiding-self-assignment-in-stdshuffle?